domains/Discussions/Tests/Feature/QuestionsIndexTest.php

```diff
@@ -0,0 +1,218 @@
+    }
+
+    /** @test */
+    public function it_list_non_deleted_question(): void
+    {
+        $deleteQuestion = QuestionFactory::new([
+            'deleted_at' => Carbon::now(),
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index'))
+            ->seeJsonStructure([
+                'data' => [
+                    [
+                        'id',
+                        'title',
+                        'slug',
+                        'description',
+                        'author',
+                        'created_at',
+                        'updated_at',
+                        'resolved_at',
+                        'deleted_at',
+                    ]
+                ],
+                'links' => [
+                    'first', 'prev', 'next', 'last',
+                ],
+            ])
+            ->seeJsonDoesntContains([
+                'email' => $deleteQuestion->author->email,
+            ])
+            ->seeJsonContains([
+                'to' => 10,
+            ]);
+    }
+
+    /** @test */
+    public function it_blocks_guest_for_many_attempts(): void
+    {
+        for ($attempt = 0; $attempt < 30; ++$attempt) {
+            $this->get(route('discussions.questions.index'))
+                ->assertResponseStatus(Response::HTTP_OK);
+        }
+
+        $this->get(route('discussions.questions.index'))
+            ->assertResponseStatus(Response::HTTP_TOO_MANY_REQUESTS);
+    }
+
+    /** @test */
+    public function it_not_blocks_authenticated_user_for_many_attempts(): void
+    {
+        $this->actingAs($this->user);
+
+        for ($attempt = 0; $attempt < 30; ++$attempt) {
+            $this->get(route('discussions.questions.index'));
+        }
+
+        $this->get(route('discussions.questions.index'))
+            ->assertResponseStatus(Response::HTTP_OK);
+    }
+
+    /** @test */
+    public function it_navigates_to_next_page(): void
+    {
+        $this->json('GET', route('discussions.questions.index', [
+            'page' => 2,
+        ]))
+            ->seeJsonContains([
+                'current_page' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_author(): void
+    {
+        $user = UserFactory::new()->create();
+        QuestionFactory::new([
+            'author_id' => $user->id,
+        ])
+            ->count(3)
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'author' => $user->id,
+        ]))
+            ->seeJsonContains([
+                'email' => $user->email,
+            ])
+            ->seeJsonContains([
+                'to' => 3,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_title(): void
+    {
+        QuestionFactory::new([
+            'title' => 'LARAVEL-PT',
+        ])->create();
+        QuestionFactory::new([
+            'title' => 'laravel-Pt',
+        ])
+            ->create();
```

Comment:
Nit: format the code by given space between code blocks, and break chained method calls into their own line. It's easier to read that way.